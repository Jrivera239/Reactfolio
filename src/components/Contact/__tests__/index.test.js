import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "..";

afterEach(cleanup);

describe("Contact Component", () => {
  it("renders", () => {
    render(<ContactForm></ContactForm>);
  });

  it("Contact snapshot", () => {
    const { asFragment } = render(<ContactForm></ContactForm>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Link visible", () => {
    it("Insert text into links", () => {
        const {getByTestId} = render(
            <ContactForm></ContactForm>
        )
        expect(getByTestId("Contact")).toHaveTextContent("Contact Me")
        expect(getByTestId("Button")).toHaveTextContent("Submit")
    })
})
