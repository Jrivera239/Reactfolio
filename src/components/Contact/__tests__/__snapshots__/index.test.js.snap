// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Contact snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="Contact"
    >
      Contact Me
    </h1>
    <form
      id="contactFrom"
    >
      <div>
        <label
          for="name"
        >
          Name: 
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email: 
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message: 
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <div>
        <p
          class="error-text"
        >
           
        </p>
      </div>
      <button
        data-testid="Button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
